homeassistant/components/tellstick/sensor.py

```diff
@@ -70,19 +75,24 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     datatype_mask = config.get(CONF_DATATYPE_MASK)
 
     for tellcore_sensor in tellcore_lib.sensors():
-        try:
-            sensor_name = config[tellcore_sensor.id]
-        except KeyError:
-            if config.get(CONF_ONLY_NAMED):
-                continue
+        if not config[CONF_ONLY_NAMED]:
             sensor_name = str(tellcore_sensor.id)
-
-        for datatype in sensor_value_descriptions:
-            if datatype & datatype_mask:
-                if tellcore_sensor.has_value(datatype):
-                    sensor_info = sensor_value_descriptions[datatype]
-                    sensors.append(TellstickSensor(
-                        sensor_name, tellcore_sensor, datatype, sensor_info))
+        else:
+            sensor_name = None
+            for named_sensor in config[CONF_ONLY_NAMED]:
```

Comment:
Break this iteration out of the tellcore sensor loop. Create a dictionary that can be used for look up.

```py
named_sensors = {
    named_sensor[CONF_ID]: named_sensor
    for named_sensor in config[CONF_ONLY_NAMED}
```

Reply:
Wow, that is so effecient.. for loop inside the dictionary declaration 😻